### PR TITLE
add bincount function

### PR DESCRIPTION
**Pull request by KsanaKozlova**

### CODE REVIEW

File: tests/test_statistics.py [hunk 1]
```diff
@@ -72,3 +72,42 @@ def test_nanvar(array):
...
+                              [2, 2, 2, 2]],
+                             ids=['[1, 2, 3]',
+                                  '[1, 2, 2, 1, 2, 4]',
+                                  '[2, 2, 2, 2]'])
+    @pytest.mark.parametrize("minlength",
+                             [0, 1, 3, 5],
+                             ids=['0', '1', '3', '5'])
+    def test_bincount_minlength(self, array, minlength):
+        a = numpy.array(array)
+        ia = dpnp.array(a)
```
Comment (samir-nasibli) [inline tests/test_statistics.py:91]:
```suggestion
        ia = dpnp.array(array)
```

File: dpnp/dpnp_iface_statistics.py [hunk 2]
```diff
@@ -169,6 +170,28 @@ def average(x1, axis=None, weights=None, returned=False):
...
+
+    See Also
+    --------
+    :obj:`dpnp.unique` : Find the unique elements of an array.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> x = np.bincount(np.arange(5))
+    >>> [i for i in x]
```
Comment (samir-nasibli) [inline dpnp/dpnp_iface_statistics.py:187]:
```suggestion
    >>> print(x)
```
Do not create a new object just for printing

File: tests/test_statistics.py [hunk 3]
```diff
@@ -72,3 +72,42 @@ def test_nanvar(array):
...
+    @pytest.mark.parametrize("weights",
+                             [None,
+                              [0.3, 0.5, 0.2, 0.7, 1., -0.6],
+                              [2, 2, 2, 2, 2, 2]],
+                             ids=['None',
+                                  '[0.3, 0.5, 0.2, 0.7, 1., -0.6]',
+                                  '[2, 2, 2, 2, 2, 2]'])
+    def test_bincount_weights(self, array, weights):
+        a = numpy.array(array)
+        ia = dpnp.array(a)
```
Comment (samir-nasibli) [inline tests/test_statistics.py:109]:
```suggestion
        ia = dpnp.array(array)
```

File: tests/test_statistics.py [hunk 4]
```diff
@@ -72,3 +72,42 @@ def test_nanvar(array):
...
+                             ids=['[1, 2, 2, 1, 2, 4]'])
+    @pytest.mark.parametrize("weights",
+                             [None,
+                              [0.3, 0.5, 0.2, 0.7, 1., -0.6],
+                              [2, 2, 2, 2, 2, 2]],
+                             ids=['None',
+                                  '[0.3, 0.5, 0.2, 0.7, 1., -0.6]',
+                                  '[2, 2, 2, 2, 2, 2]'])
+    def test_bincount_weights(self, array, weights):
+        a = numpy.array(array)
```
Comment (samir-nasibli) [inline tests/test_statistics.py:108]: avoid using single-letter variable names. `a`-> `np_a`.
In this case `ai` -> `dpnp_a` is better.

**Review by densmirn — APPROVED** [review]

File: dpnp/dpnp_iface_statistics.py [hunk 5]
```diff
@@ -59,7 +60,7 @@
     'min',
     'nanvar',
     'std',
-    'var',
+    'var'
```
Comment (densmirn) [inline dpnp/dpnp_iface_statistics.py:63]: I would keep comma here. When inserting new string to the end of the list you can accidently forget to add the comma that can be a potential issue.

**Review by shssf — APPROVED** [review]

**Pull request merged**